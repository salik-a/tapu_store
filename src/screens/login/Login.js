import React, { useState, useEffect } from "react";
import { ButtonsContainer, Container, Header, TopContainer, Form, Picker, Name, Text } from './LoginStyled'
import Input from "../../components/input/Input"
import Button from "../../components/button/Button"
import { Formik } from "formik";
import auth from "@react-native-firebase/auth"
import DropDownPicker from 'react-native-dropdown-picker';


const Login = ({ navigation }) => {
    const [user, setUser] = useState(auth().currentUser);
    const [open, setOpen] = useState(false);
    const [value, setValue] = useState("TR");

    const items = [
        { label: 'Türkçe', value: 'TR' },
        { label: 'English', value: 'EN' }
    ];

    const initialForm = {
        usermail: "",
        password: ""
    }

    const handleLogin = async (formValues) => {
        try {
            await auth().signInWithEmailAndPassword(formValues.usermail, formValues.password)
            setUser(auth().currentUser)
        } catch (error) {
            console.log(error)
        }
    }
    const handleLogOut = async () => {
        try {
            await auth().signOut()
            console.log("signed out");
            setUser(null)
        } catch (error) {
            console.log(error)
        }
    }
    console.log(user)
    return (
        <Container >
            <TopContainer>
                <Header>Account</Header>
            </TopContainer>

            {user === null ? (
                <Formik
                    initialValues={initialForm}
                    onSubmit={values => handleLogin(values)}
                >
                    {({ handleSubmit, values, handleChange }) => (
                        <Form>
                            <Input
                                placeholder="E-Mail"
                                onChangeText={handleChange('usermail')}
                                value={values.usermail}
                            />
                            <Input
                                placeholder="Password"
                                onChangeText={handleChange('password')}
                                value={values.password}
                                isSecure={true}
                            />
                            <Picker>
                                <DropDownPicker
                                    open={open}
                                    value={value}
                                    items={items}
                                    setOpen={() => setOpen(!open)}
                                    setValue={setValue}

                                    dropDownContainerStyle={{
                                        width: "100%",
                                        alignSelf: 'center',
                                        position: 'relative',
                                        top: 0,
                                        borderColor: "white"
                                    }}
                                    listMode="SCROLLVIEW"
                                    style={{
                                        height: 40,
                                        alignSelf: 'center',
                                        marginTop: 10,
                                        backgroundColor: "white",
                                        borderColor: "white",

                                    }}
                                    labelStyle={{
                                        fontSize: 16,
                                    }}

                                />
                            </Picker>
                            <ButtonsContainer>
                                <Button title="Sign In" onPress={handleSubmit} bgColor='#E82223' />
                                <Button title="Sign Up" onPress={() => navigation.navigate("SignUp")} bgColor='#BBC3CF' />
                            </ButtonsContainer>
                        </Form>
                    )}
                </Formik>
            ) : (
                <Container>
                    <TopContainer>
                            <Name>{user.displayName}</Name>
                            <Text>E-mail: {user.email}</Text>
                            <Text>Password: ******</Text>
                        <Text>Current locale: {value} </Text>
                    </TopContainer>
                    <Picker>
                        <DropDownPicker
                            open={open}
                            value={value}
                            items={items}
                            setOpen={() => setOpen(!open)}
                            setValue={setValue}

                            dropDownContainerStyle={{
                                width: "100%",
                                alignSelf: 'center',
                                position: 'relative',
                                top: 0,
                                borderColor: "white"
                            }}
                            listMode="SCROLLVIEW"
                            style={{
                                height: 40,
                                alignSelf: 'center',
                                marginTop: 10,
                                backgroundColor: "white",
                                borderColor: "white",

                            }}
                            labelStyle={{
                                fontSize: 16,
                            }}

                        />
                    </Picker>
                    <ButtonsContainer>
                            <Button title="Log Out" onPress={handleLogOut} bgColor='#BBC3CF' textColor borderColor />

                    </ButtonsContainer>
                </Container>
            )}

        </Container>
    );
};

export default Login;