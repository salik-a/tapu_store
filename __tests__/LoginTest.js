jest.mock('@react-native-firebase/auth', () => {
    return () => ({
        signInWithEmailAndPassword: jest.fn(),
        createUserWithEmailAndPassword: jest.fn(),
        signOut: jest.fn(),

    });
});
import React from 'react'
import { fireEvent, render, waitFor } from '@testing-library/react-native'
import Login from '../src/screens/login/Login';



test('should match with snapshot', () => {

    const comp = render(<Login />);
    expect(comp).toMatchSnapshot();
});

