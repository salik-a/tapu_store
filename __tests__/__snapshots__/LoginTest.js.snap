// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "marginBottom": 15,
          "marginTop": 25,
          "width": "85%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 30,
            "fontWeight": "800",
            "marginVertical": 10,
          },
        ]
      }
    >
      Account
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "marginBottom": 15,
            "marginTop": 25,
            "width": "85%",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 34,
              "fontWeight": "bold",
              "marginBottom": 20,
            },
          ]
        }
      >
        test test
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 16,
              "marginVertical": 2,
            },
          ]
        }
      >
        E-mail: 
        test@gmail.com
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 16,
              "marginVertical": 2,
            },
          ]
        }
      >
        Password: ******
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 16,
              "marginVertical": 2,
            },
          ]
        }
      >
        Current locale: 
        TR
         
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "borderBottomWidth": 2,
            "borderColor": "#64738C",
            "width": "85%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "width": "100%",
            },
            Object {
              "zIndex": 5000,
            },
            Object {},
            false,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onLayout={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "white",
              "borderColor": "white",
              "borderRadius": 8,
              "borderWidth": 1,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "space-between",
              "marginTop": 10,
              "opacity": 1,
              "paddingHorizontal": 10,
              "width": "100%",
              "zIndex": 5000,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "flex": 1,
                },
                Object {},
                Object {
                  "fontSize": 16,
                },
                false,
              ]
            }
          >
            Türkçe
          </Text>
          <View
            style={
              Array [
                Object {
                  "marginLeft": 10,
                },
                Object {},
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../react-native-dropdown-picker/src/themes/light/icons/arrow-down.png",
                }
              }
              style={
                Array [
                  Object {
                    "height": 20,
                    "width": 20,
                  },
                  Object {},
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "flex-end",
            "marginBottom": 60,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "white",
            "borderColor": "#E82223",
            "borderRadius": 12,
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "marginVertical": 10,
            "opacity": 1,
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
            "width": "85%",
          }
        }
        testID="button-touchable"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#E82223",
                "fontSize": 14,
                "textAlign": "center",
              },
            ]
          }
          testID="button-title"
          textColor="#E82223"
        >
          Log Out
        </Text>
      </View>
    </View>
  </View>
</View>
`;
